### DISCUSSION TITLE
0.20.0 with new statistics!

### DISCUSSION THREAD

**Discussion by meikpiep** [meikpiep/gauguin#76] [Announcements]
# Added

- When solving a game, the keypad gets swapped by a card view which states the win of a game. It
  also shows if it was the first game solved ever, or the first kind been solved, or a best time
  which could be made.
- Added more detailed statistics. After solving at least one game, three diagrams visualize
  - The difficulty of the games solved.
  - The play time of the games solved.
  - The streak history of all games.
  There is currently no statistics by game variant. Feedback warmly welcome.

# Changed

- Move the popup messages of the main screen to the absolute bottom of the screen, now covering the
 bottom app bar. Avoids covering the bottom row of number buttons when checking if there are errors.
- Mark 'Fast Finishing Mode' as stable.
- Show hours of play time only if the game lasts longer than one hour.

# Removed

- Removed message when cheating/revealing cells. It seemed to be not motivating at all.
- Removed the possibility to shift the keypad from center to the left and right.

<hr /><em>This discussion was created from the release <a href='https://github.com/meikpiep/gauguin/releases/tag/v0.20.0'>0.20.0</a>.</em>

**Comment by hypr1** [comment]
Don't the statistics end up being a bit pointless since difficulty and duration both change drastically even between very easy modes because of the size? (Keeping track of streaks is pretty interesting but the graph goes through decimals, understandable but still odd when the only values are non-negative integers)

**Reply by meikpiep** [reply]
That's correct. I intend to supply statistics dependent on the game variant, but I could not figure out a system which is variant dependent without having a bunch of lengthy statistics in mind. That's why I started with some kind of overall statistics.

The tricky part for me is the question: Which statistics would enable interesting insides - would it be exclusively the variant last played? Would the user want to see all variants he solved historically? To what detail? What are the interesting details?

The only way I could think of would be to supply diagrams analogue to the ones already implemented to all variants and to store and display them variant dependent and to emphasize both the overall ones and the current variant. To be honest, I have more questions as answers.

Feel free to add any thoughts, any of you. It is a cool way to gather new perspectives from such discussions.

**Comment by meikpiep** [comment]
FYI: Version 0.20.0 just arrived via F-Droid.
